src/protocols/interaction_model/StatusCode.h

```diff
@@ -48,6 +50,108 @@ enum class Status : uint8_t
+        this->mClusterSpecificCode = chip::NullOptional;
+        return *this;
+    }
+
+    /**
+     * @brief Builder for a cluster-specific failure status code.
+     *
+     * @tparam T - enum type for the cluster-specific status code
+     *             (e.g. chip::app::Clusters::AdministratorCommissioning::CommissioningWindowStatusEnum)
+     * @param cluster_specific_code - cluster-specific code to record with the failure
+     *             (e.g. chip::app::Clusters::AdministratorCommissioning::CommissioningWindowStatusEnum::kWindowNotOpen)
+     * @return a StatusCode instance properly configured.
+     */
+    template <typename T>
+    static StatusCode ClusterSpecificFailure(T cluster_specific_code)
+    {
+        static_assert(std::numeric_limits<T>::max() <= std::numeric_limits<uint8_t>::max(), "Type used must fit in uint8_t");
+        return StatusCode(Status::Failure, static_cast<uint8_t>(cluster_specific_code));
+    }
+
+    /**
+     * @brief Builder for a cluster-specific success status code.
+     *
+     * @tparam T - enum type for the cluster-specific status code
+     *             (e.g. chip::app::Clusters::AdministratorCommissioning::CommissioningWindowStatusEnum)
+     * @param cluster_specific_code - cluster-specific code to record with the success
+     *             (e.g. chip::app::Clusters::AdministratorCommissioning::CommissioningWindowStatusEnum::kBasicWindowOpen)
+     * @return a StatusCode instance properly configured.
+     */
+    template <typename T>
+    static StatusCode ClusterSpecificSuccess(T cluster_specific_code)
+    {
+        static_assert(std::numeric_limits<T>::max() <= std::numeric_limits<uint8_t>::max(), "Type used must fit in uint8_t");
+        return StatusCode(Status::Success, static_cast<uint8_t>(cluster_specific_code));
+    }
+
+    /// @return true if the core Status associated with this StatusCode is the one for success.
+    bool IsSuccess() const { return mStatus == Status::Success; }
+
+    /// @return the core Status code associated withi this StatusCode.
+    Status GetStatus() const { return mStatus; }
+
+    /// @return true if a cluster-specific code is associated with the StatusCode.
+    bool HasClusterSpecificCode() const { return mClusterSpecificCode.HasValue(); }
+
+    /// @return the cluster-specific code associated with this StatusCode or chip::NullOptional if none is associated.
+    chip::Optional<uint8_t> GetClusterSpecificCode() const
+    {
+        if (!mClusterSpecificCode.HasValue() || !((mStatus == Status::Failure) || (mStatus == Status::Success)))
+        {
+            return chip::NullOptional;
+        }
+        return mClusterSpecificCode;
+    }
+
+    // Automatic conversions to common types, using the status code alone.
+    operator Status() const { return mStatus; }
+    operator int() const { return static_cast<int>(mStatus); }
+    operator uint8_t() const { return static_cast<uint8_t>(mStatus); }
```

Comment:
These are a bit misuse-prone, it seems to me, but ok...

Reply:
What is misuse-prone about it?

Reply:
Jsut removed them since not really used.

Reply:
Just people losing type info and starting to pass integers around....  Basically would prefer that to be explicit, not implicit.